### ISSUE TITLE
Exception with pinned tweet: TypeError: string indices must be integers

### ISSUE THREAD

**Issue by nemobis** [robertoszek/pleroma-bot#69]
While posting to https://respublicae.eu/@Europarl_FI for the first time, I got an exception upon posting and pinning https://respublicae.eu/@Europarl_FI/108067690963226835 from https://nitter.eu/Europarl_FI/status/1502279955908071425 (even though it seems to have worked). 

I'm currently using 1.0.3rc6

```
ℹ 2022-04-03 13:26:50,744 - pleroma_bot - INFO - ====================================== 
ℹ 2022-04-03 13:26:50,744 - pleroma_bot - INFO - Processing user:       108067243532704868 
ℹ 2022-04-03 13:26:50,745 - pleroma_bot - INFO - It seems like pleroma-bot is running for the first time for this Twitter user: Europarl_FI 
⚠ 2022-04-03 13:26:52,241 - pleroma_bot - WARNING - Mastodon doesn't support display names longer than 30 characters, truncating it and trying again... (_utils.py:501) 
ℹ 2022-04-03 13:26:52,241 - pleroma_bot - INFO - How far back should we retrieve tweets from the Twitter account? 
ℹ 2022-04-03 13:26:52,242 - pleroma_bot - INFO - 
Enter a date (YYYY-MM-DD):
[Leave it empty to retrieve *ALL* tweets or enter 'continue'
if you want the bot to execute as normal (checking date of 
last post in the Fediverse account)]  
⚠ 2022-04-03 13:26:52,531 - pleroma_bot - WARNING - No posts were found in the target Fediverse account (_pleroma.py:43) 
ℹ 2022-04-03 13:26:52,532 - pleroma_bot - INFO - How far back should we retrieve tweets from the Twitter account? 
ℹ 2022-04-03 13:26:52,532 - pleroma_bot - INFO - 
Enter a date (YYYY-MM-DD):
[Leave it empty to retrieve *ALL* tweets or enter 'continue'
if you want the bot to execute as normal (checking date of 
last post in the Fediverse account)]  
ℹ 2022-04-03 13:26:53,440 - pleroma_bot - INFO - tweets gathered:        9 
ℹ 2022-04-03 13:26:59,471 - pleroma_bot - INFO - tweets to post:         9 
ℹ 2022-04-03 13:26:59,472 - pleroma_bot - INFO - (1/9) 
ℹ 2022-04-03 13:27:02,070 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:02,571 - pleroma_bot - INFO - (2/9) 
ℹ 2022-04-03 13:27:02,912 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:03,413 - pleroma_bot - INFO - (3/9) 
ℹ 2022-04-03 13:27:05,887 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:06,388 - pleroma_bot - INFO - (4/9) 
ℹ 2022-04-03 13:27:09,143 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:09,644 - pleroma_bot - INFO - (5/9) 
ℹ 2022-04-03 13:27:10,116 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:10,617 - pleroma_bot - INFO - (6/9) 
ℹ 2022-04-03 13:27:13,170 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:13,671 - pleroma_bot - INFO - (7/9) 
ℹ 2022-04-03 13:27:16,168 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:16,669 - pleroma_bot - INFO - (8/9) 
ℹ 2022-04-03 13:27:19,685 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:20,186 - pleroma_bot - INFO - (9/9) 
ℹ 2022-04-03 13:27:23,307 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:23,808 - pleroma_bot - INFO - Current pinned:        1502279955908071425 
ℹ 2022-04-03 13:27:23,809 - pleroma_bot - INFO - Previous pinned:       None 
ℹ 2022-04-03 13:27:26,853 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:27:26,854 - pleroma_bot - INFO - File with previous pinned post ID not found or empty. Checking last posts for pinned post... 
✖ 2022-04-03 13:27:26,854 - pleroma_bot - ERROR - Exception occurred (cli.py:539) 
Traceback (most recent call last):
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/cli.py", line 524, in main
    user.check_pinned(posted)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_utils.py", line 367, in check_pinned
    pleroma_pinned_post = self.pin_pleroma(id_post_to_pin)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_pin.py", line 127, in pin_pleroma
    self.unpin_pleroma(pinned_file)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_pin.py", line 173, in unpin_pleroma
    _find_pinned(self, pinned_file)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_pin.py", line 188, in _find_pinned
    if post["pinned"]:
TypeError: string indices must be integers
```

**Comment by nemobis** [comment]
Another one:
```
ℹ 2022-04-03 13:35:08,271 - pleroma_bot - INFO - ====================================== 
ℹ 2022-04-03 13:35:08,271 - pleroma_bot - INFO - Processing user:       108067315577208905 
ℹ 2022-04-03 13:35:08,272 - pleroma_bot - INFO - It seems like pleroma-bot is running for the first time for this Twitter user: eucopresident 
ℹ 2022-04-03 13:35:10,628 - pleroma_bot - INFO - How far back should we retrieve tweets from the Twitter account? 
ℹ 2022-04-03 13:35:10,629 - pleroma_bot - INFO - 
Enter a date (YYYY-MM-DD):
[Leave it empty to retrieve *ALL* tweets or enter 'continue'
if you want the bot to execute as normal (checking date of 
last post in the Fediverse account)]  
⚠ 2022-04-03 13:35:11,000 - pleroma_bot - WARNING - No posts were found in the target Fediverse account (_pleroma.py:43) 
ℹ 2022-04-03 13:35:11,001 - pleroma_bot - INFO - How far back should we retrieve tweets from the Twitter account? 
ℹ 2022-04-03 13:35:11,001 - pleroma_bot - INFO - 
Enter a date (YYYY-MM-DD):
[Leave it empty to retrieve *ALL* tweets or enter 'continue'
if you want the bot to execute as normal (checking date of 
last post in the Fediverse account)]  
ℹ 2022-04-03 13:35:11,909 - pleroma_bot - INFO - tweets gathered:        5 
ℹ 2022-04-03 13:35:13,124 - pleroma_bot - INFO - tweets to post:         5 
ℹ 2022-04-03 13:35:13,124 - pleroma_bot - INFO - (1/5) 
ℹ 2022-04-03 13:35:13,767 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:35:14,269 - pleroma_bot - INFO - (2/5) 
ℹ 2022-04-03 13:35:14,623 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:35:15,124 - pleroma_bot - INFO - (3/5) 
ℹ 2022-04-03 13:35:15,488 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:35:15,990 - pleroma_bot - INFO - (4/5) 
ℹ 2022-04-03 13:35:16,358 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:35:16,860 - pleroma_bot - INFO - (5/5) 
ℹ 2022-04-03 13:35:17,250 - pleroma_bot - INFO - Post in Pleroma:       <Response [200]> 
ℹ 2022-04-03 13:35:17,752 - pleroma_bot - INFO - Current pinned:        1510542585454022660 
ℹ 2022-04-03 13:35:17,752 - pleroma_bot - INFO - Previous pinned:       None 
ℹ 2022-04-03 13:35:17,752 - pleroma_bot - INFO - File with previous pinned post ID not found or empty. Checking last posts for pinned post... 
✖ 2022-04-03 13:35:17,752 - pleroma_bot - ERROR - Exception occurred (cli.py:539) 
Traceback (most recent call last):
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/cli.py", line 524, in main
    user.check_pinned(posted)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_utils.py", line 367, in check_pinned
    pleroma_pinned_post = self.pin_pleroma(id_post_to_pin)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_pin.py", line 127, in pin_pleroma
    self.unpin_pleroma(pinned_file)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_pin.py", line 173, in unpin_pleroma
    _find_pinned(self, pinned_file)
  File "/home/federico/mw/pleroma-bot/res/lib64/python3.10/site-packages/pleroma_bot/_pin.py", line 188, in _find_pinned
    if post["pinned"]:
TypeError: string indices must be integers
```

**Comment by robertoszek** [comment]
Looks like it's happening when running it for the first time for an user which has no posts, right?

I'm thinking it's trying to find a previous pinned post in an empty list that somehow it didn't evaluate to false.
I need to check what `/api/v1/accounts/<user_id>/statuses` returns for an empty user on a Mastodon instance, it looks like it's getting actually a string for whatever reason.

**Comment by nemobis** [comment]
Il 03/04/22 17:45, robertoszek ha scritto:
> Looks like it's happening when running it for the first time for an user which has no posts, right?

Indeed. The second time it usually succeeds, so the end result is that 
the pinned tweet is posted twice but then everything works. Not a huge 
deal, just a bit noisy.

**Comment by robertoszek** [comment]
I see now how this could be triggered on first run with an empty account,
it should be fixed on 1.0.3rc8, feel free to give it a spin and see if the changes solved the issue:
```
pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pleroma-bot==1.0.3rc8
```

**Comment by nemobis** [comment]
Il 03/04/22 20:21, robertoszek ha scritto:
> it should be fixed on 1.0.3rc8, feel free to give it a spin and see if the changes solved the issue:

Looks fixed indeed, thank you very much! Tested with over 50 users, 
didn't appear again.

**Comment by robertoszek** [comment]
Awesome! Thanks for reporting back, glad it works as expected now.
Feel free to open a new issue if you run into one, have a nice day!
